Fix:

error: implicitly declaring library function 'strncasecmp' with type 'int (const char *, const char *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
Copied from the php70 ports

https://github.com/php/php-src/commit/1a8df11364292dd3cac2c406a26e4aafaca62d36
--- ext/curl/config.m4.orig	2019-01-09 09:54:13.000000000 +0000
+++ ext/curl/config.m4	2021-07-22 18:17:39.000000000 +0100
@@ -61,6 +61,7 @@
     AC_PROG_CPP
     AC_MSG_CHECKING([for openssl support in libcurl])
     AC_TRY_RUN([
+#include <strings.h>
 #include <curl/curl.h>
 
 int main(int argc, char *argv[])
@@ -88,6 +89,7 @@
    
     AC_MSG_CHECKING([for gnutls support in libcurl])
     AC_TRY_RUN([
+#include <strings.h>
 #include <curl/curl.h>
 
 int main(int argc, char *argv[])
